src/ir/possible-contents.h

```diff
@@ -199,8 +221,10 @@ class PossibleContents {
       return size_t(1) | (std::hash<Literal>()(getLiteral()) << 3);
     } else if (isGlobal()) {
       return size_t(2) | (std::hash<Name>()(getGlobal()) << 3);
-    } else if (isExactType()) {
-      return size_t(3) | (std::hash<Type>()(getType()) << 3);
+    } else if (auto* coneType = std::get_if<ConeType>(&value)) {
+      return size_t(3) | ((std::hash<std::pair<Type, Index>>{}(
+                            {coneType->type, coneType->depth}))
+                          << 3);
```

Comment:
Instead of `|` and the shift, this should probably be using `wasm::rehash`. If the distribution of content kinds is not uniform, the low two bits of the hash will have less than two bits of entropy with this strategy.

Reply:
Makes sense, yeah, this is worth a separate improvement. I'll add a TODO for myself.